controllers/drplacementcontrol_controller.go

```diff
@@ -331,31 +390,30 @@ func (r *DRPlacementControlReconciler) Reconcile(ctx context.Context, req ctrl.R
 		return ctrl.Result{}, errorswrapper.Wrap(err, "failed to get DRPC object")
 	}
 
-	usrPlRule, err := r.getUserPlacementRule(ctx, drpc, logger)
+	placementObj, err := r.getPlacementOrPlacementRule(ctx, drpc, logger)
 	if err != nil {
-		r.recordFailure(drpc, usrPlRule, "Error", err.Error(), nil, logger)
+		r.recordFailure(drpc, placementObj, "Error", err.Error(), nil, logger)
 
 		return ctrl.Result{}, err
 	}
 
-	// If either drpc or User PlacementRule is deleted, then we must cleanup.
-	if r.isBeingDeleted(drpc, usrPlRule) {
-		// DPRC depends on User PlacementRule. If DRPC or/and the User PlacementRule is deleted,
+	if r.isBeingDeleted(drpc) {
```

Comment:
Checking for userPl being deleted and triggering a DRPC cleanup was required as in certain cases the userPl and the resources pointing to the userPl were deleted, which if not processed made DRPC hold back the PVCs etc. causing garbage collection to stall till DRPC was deleted as well (to enter the deletion routines).

Reply:
Ah, I removed it because I couldn't remember why we had that.  It just didn't make sense to me. Now I remember.